app/services/data_migrations/remove_application_choice_in_the_incorrect_cycle.rb

```diff
@@ -1,16 +1,15 @@
 module DataMigrations
-  class RemovePreviousCyclesCoursesFromApplicationsInTheCurrentCycle
-    TIMESTAMP = 20210528131818
+  class RemoveApplicationChoiceInTheIncorrectCycle
+    TIMESTAMP = 20210602144637
     MANUAL_RUN = false
 
     def change
       ApplicationChoice
       .joins(:application_form, course_option: [:course])
-      .where(course: { recruitment_cycle_year: RecruitmentCycle.previous_year })
-      .where(application_form: { recruitment_cycle_year: RecruitmentCycle.current_year })
+      .where('courses.recruitment_cycle_year != application_forms.recruitment_cycle_year')
       .find_each(batch_size: 100) do |application_choice|
         application_choice.application_form.update!(
-          audit_comment: "Application_choice ##{application_choice.id} was deleted due to being associated with a course from last years recruitment cycle",
+          audit_comment: "Application_choice ##{application_choice.id} was deleted due to being associated with a course from another recruitment cycle",
         )
         application_choice.destroy
```

Comment:
This looks good. Shall we have a dry run option, or at least an informal route to dry run this (eg expose the query as a method on the class and make it a manual migration)? Just in _case_ we make a mistake

Reply:
Will do. Just closing this for now